src/Type/Php/ExplodeFunctionDynamicReturnTypeExtension.php

```diff
@@ -59,6 +60,8 @@ public function getTypeFromFunctionCall(
 		$stringType = $scope->getType($args[1]->value);
 		if ($stringType->isLowercaseString()->yes()) {
 			$returnValueType = new IntersectionType([new StringType(), new AccessoryLowercaseStringType()]);
+		} elseif ($stringType->isUppercaseString()->yes()) {
```

Comment:
It should not be elseif.

https://phpstan.org/r/4d463db9-c5ce-43a2-8261-b00274381327

```
explode('-', '01-23-45')
```
should give `non-empty-list<lowercase-string&uppercase-string>`

but will give `non-empty-list<lowercase-string>` because of the order of the condition here.

code should be
```
$accessory = [];
if ($stringType->isLowercaseString()->yes()) {
    $accessory[] = new AccessoryLowercaseStringType();
} 
if ($stringType->isUppercaseString()->yes()) {
    $accessory[] = new AccessoryUppercaseStringType();
}
if (count($accessory) > 0) {
    $returnValueType = new IntersectionType([new StringType(), ...$accessory]);
} else {
    $returnValueType = new StringType();
}

Reply:
Fixed in 3c6a7b3a3 -- note the modifications to the test cases as well.